lib/entities/game.rb

```diff
@@ -3,38 +3,38 @@
 MESSAGE_GU = { "attempt": 'user_answer',
                "nil": 'nil',
                "win": 'win', "failure": 'failure' }.freeze
-PLUS = '+'
-MINUS = '-'
+PLUS = '+'.freeze
+MINUS = '-'.freeze
 
 class Game
   attr_reader :difficulty, :secret_code
-  attr_accessor :hints_total, :hint_clone_scode, :current_hint
+  attr_accessor :hint_clone_scode
 
   def initialize(difficulty)
     @difficulty = difficulty
-    @hints_total = difficulty[:difficulty][:hints].to_i
     @secret_code = create_secret_code
     @hint_clone_scode = @secret_code.shuffle
   end
 
   def compare(user_input)
-    MESSAGE_GU[:win] if user_input.chars == @secret_code
-    plus_minus_factoring(user_input) unless user_input.chars == @secret_code
+    if user_input.chars == @secret_code
+      MESSAGE_GU[:win]
+    else
+      plus_minus_factoring(user_input)
+    end
   end
 
-  def view_hint(current_hint)
-    @current_hint = current_hint
+  def hint
     remainder = @hint_clone_scode.pop(@hint_clone_scode.length - 1)
-    hint = (@hint_clone_scode - remainder)
-    @current_hint -= 1
+    hint = @hint_clone_scode - remainder
```

Comment:
by a request to instance variable ` @hint_clon_scode `  and `@current_hint from Rspec ` response is  NIL. Code RSpec is in file game_spec.rb:117:

Reply:
you can test that the number of hints will be changed like this:
```
context 'when the hint was viewed' do
    it 'reduces the number of hints by 1' do
      expect { game.hint }.to change { game.hint_clone_scode.size }.by(-1)
    end
  end
```